building-and-packaging/material/packaging_python_slides.md

```diff
@@ -0,0 +1,294 @@
+
+```bash
+generic_folder_name/
+├── LICENSE
+├── setup.py
+├── README.md
+├── src/
+│   └── package_name/
+│       ├── __init__.py
+│       └── solver.py
+└── tests/
+```
+
+<span>
+Lets have a detailed look at individual files in this folder structure
+<!-- .element: class="fragment" data-fragment-index="1" --></span>
+
+---
+
+## Additional options in setup.py 1/2
+
+- [Classifiers](https://pypi.org/classifiers/): additional metadata for the version of the package
+- Defined as part of [PEP 303](https://www.python.org/dev/peps/pep-0301/#distutils-trove-classification)
+- Example:
+
+```bash
+from setuptools import setup
+
+setup(
+    ...
+    classifiers=[
+        "Programming Language :: Python :: <Python-version>",
+        "License :: OSI Approved :: <Name of License>",
+        "Operating System :: <OS's on which this package runs>",
+    ],
+    ...
+)
+```
+
+---
+
+## Additional options in setup.py 2/2
+
+- The file `README.md` can be passed as a long description in the following way:
+
+```bash
+from setuptools import setup
+
+with open("README.md", "r", encoding="utf-8") as fh:
+    long_description = fh.read()
+
+setup(
+    ...
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    ...
+)
+```
+
+---
+
+## LICENSE
+
+- This file provides details on which license is the package distributed under
+- Choosing the correct license is often a critical step in open-source software development
+- More on licenses in a future chapter of this course
+
+---
+
+## README.md
+
+- A `README.md` file typically gives a description of the package, code structure, meta information, the installation procedure and more ...
+- It is common practice to parse the `README.md` to `setuptools` as a long description
+- Newer versions of `setuptools` also include this file automatically
+- This file can also be written using [reStructuredText](https://docutils.sourceforge.io/rst.html) which is part of [Docutils](https://docutils.sourceforge.io/index.html)
+
+---
+
+## Creating distribution archives
+
+- The project needs to be converted to distribution archives so that pip can install them
+- There are two ways to create these archives:
+    - Using the package builder [build](https://pypa-build.readthedocs.io/en/stable/index.html) which was introduced in [PEP 517](https://www.python.org/dev/peps/pep-0517/)
+    - Using the packaging standard [wheel](https://wheel.readthedocs.io/en/stable/) which was introduced in [PEP 427](https://www.python.org/dev/peps/pep-0427/)
+- A source distribution archive file (`package-name-<tags>.tar.gz`) can be generated by running the command `python3 setup.py sdist` in the `package_name/` repository
+- A wheel archive file (`package-name-<tags>.whl`) can be generated by running the command `python3 setup.py bdist_wheel` in the `package_name/` repository
+- In both cases the files will be generated in a folder `package_name/dist/`
+
```

Comment:
I guess you will explain the difference between the two package types?! I am very curious at least to learn about the difference and reasons for different package formats.